### ISSUE TITLE
[gitea] rework some guide steps

### ISSUE THREAD

**Issue by noave** [Uberspace/lab#1219]
- the pgp verfification is no longer working, update it or maybe just leave it
- choose a single installation way, not a "minimum set and could be done via web"
- move the external renderer to a later configuration option, and explain what it is for
- do not create the admin user password on console, let the user do it her way
- also put the ssh key feature to later configuration
- do not provide subfolder installation, keept it simple
- add startsecs=30 to service config

**Comment by EV21** [comment]
> * the pgp verification is no longer working, update it or maybe just leave it

I already fixed that in my PR #1215 . Do you think we don't need it for the manual install? Maybe just keep it for the update script?

---
> * choose a single installation way, not a "minimum set and could be done via web"

Agreed, I never used the web `/install` way before. ... checking on that, maybe it generates a more useful config file ... 🤔 

* I did one run and it looks like it is not possible to configure sendmail this way, just SMTP
* It also prefills a wrong root url with the internal port and it spams many default values into the app.ini file.

I'll check if we can avoid using the app.ini in favour to environment variables we can just set via the `~/etc/services.d/gitea.ini`, so the user does not need to edit 8 different lines within the app.ini file.

---
> * move the external renderer to a later configuration option, and explain what it is for

Yes, that should be somewhere behind the "finishing" line.

---
> * do not create the admin user password on console, let the user do it her way

Is there another way?
Or do you mean to replace that gen-thing?

```
[isabell@stardust ~]$ ADMPWD=$(pwgen 16 1)
[isabell@stardust ~]$ ADMPWD=SuperSecretAdminPassword
```

---
> * also put the ssh key feature to later configuration

somewhere behind the "finishing" line...

---
> * do not provide subfolder installation, keept it simple

👍

---
> * add startsecs=30 to service config

Yes, at startup Gitea might run some migration and optimisation steps (the same as the `gitea migrate` command). And sometimes it stops its operation after big upgrades, so `autorestart=yes` could also be a thing. Is that actually a default setting on uberspace or do we have to set this?

---
* The default Port is actually 3000, so I think we can delete `HTTP_PORT = 9000` from the config file and just use 3000 for the backend
* some of the secret keys are automatically generated when they are needed like the `INTERNAL_TOKEN` (at first startup), the `LFS_JWT_SECRET` (when LFS server is started), `JWT_SECRET` is only used if there is a not default signing algorithm configured - I'm not sure about the `SECRET_KEY` maybe just generate this one (the web installer does not generate this one) and ignore the other ones.

**Comment by noave** [comment]
thanks for this :slightly_smiling_face:  I did not see your PR before, all further discussion in  https://github.com/Uberspace/lab/pull/1215
